modules/mono/glue/cs_files/Basis.cs

```diff
@@ -155,34 +173,41 @@ public Vector3 GetAxis(int axis)
             return new Vector3(this[0, axis], this[1, axis], this[2, axis]);
         }
 
+        public void SetAxis(int axis, Vector3 value)
+        {
+            this[0, axis] = value.x;
+            this[1, axis] = value.y;
+            this[2, axis] = value.z;
+        }
+
         public Vector3 GetEuler()
         {
             Basis m = this.Orthonormalized();
 
             Vector3 euler;
             euler.z = 0.0f;
 
-            float mxy = m.y[2];
+            float mxy = m[1, 2];
 
 
             if (mxy < 1.0f)
             {
                 if (mxy > -1.0f)
                 {
                     euler.x = Mathf.Asin(-mxy);
-                    euler.y = Mathf.Atan2(m.x[2], m.z[2]);
-                    euler.z = Mathf.Atan2(m.y[0], m.y[1]);
+                    euler.y = Mathf.Atan2(m[0, 2], m[2, 2]);
+                    euler.z = Mathf.Atan2(m[1, 0], m[1, 1]);
                 }
                 else
                 {
                     euler.x = Mathf.PI * 0.5f;
-                    euler.y = -Mathf.Atan2(-m.x[1], m.x[0]);
+                    euler.y = -Mathf.Atan2(-m[0, 1], m[0, 0]);
                 }
             }
             else
             {
                 euler.x = -Mathf.PI * 0.5f;
-                euler.y = -Mathf.Atan2(m.x[1], m.x[0]);
+                euler.y = -Mathf.Atan2(-m[0, 1], m[0, 0]);
```

Comment:
I'm wondering, is `[0, ...]` supposed to return `x` or `_x`? I mention it because if that is the case and it's fixed in the future, this code would be broken.

Reply:
Yeah thats a good question.  I *think* @tagcup would say that m[0] should be rows (_x) and not columns (x) based on our conversation in #16937, but ill let him/her chime in.

As this is the same class, we could also use m._x[1] instead of m[0,1]

Reply:
I think we should also replace _x, _y and _z with 9 floats that are named accordingly to the row and column they represent, just like [OpenTK's Matrix3](https://github.com/mono/opentk/blob/master/Source/OpenTK/Math/Matrix3.cs#L38-L63).

Reply:
Thats definitely less confusing and it has my vote too.  But I also vote that we unbreak master with the current change first.  Its been in a bad state for awhile now and I think fixing it is more important than implementation details.

Reply:
I agree. There are two conflicts after the last PR I merged. It can be merged after that.

Reply:
Cool.  It should be good to go now

Reply:
Unless the i in m[i][j] refers to rows, you'll get a bunch of confused users and probably PRs with mistakes in matrix indices.

@reduz was saying OpenGL reverses indices in maths so it won't change I'm law here blah blah but it's totally irrelevant because 1) that's not true today (OpenGL accepts both versions) 2) that'd only be relevant if you're going to write a new renderer under drivers/ where godot's matrix is converted into OpenGL's matrix so has nothing to do with ordinary users.

I agree that it's probably best to use mij with i,j=1,2,3 (or 0,1,2) and/or overload a two-index array operator and exclusively use it in all the code.

Reply:
i does currently refer to rows (see my other comment)

Reply:
I have no problem with using mij (ex: m01) or something similar.  But R0C1 is unambiguous so I dont have a problem with that either

Reply:
Mij is also unambiguous unless you skipped high school

Reply:
or see the top picture on wikipedia for example https://en.wikipedia.org/wiki/Matrix_(mathematics)

Reply:
What I'm trying to say is, there is no such thing as column-major in written math. It's just a programming language thing regarding how multidimensional array operators access the underlying data in memory. *Naming* a variable Mij has nothing to do with those operators, it'd just be its mathematical name.

Reply:
and mathematical notation is always row-major

Reply:
I wasn't trying to make a point about mij, just that R0C1 isnt any worse from an information presentation standpoint.  I'm 100% ok with mij.